### ISSUE TITLE
Changes to Che-Theia preferences are persisted and loaded at every workspace start using a ConfigMap

### ISSUE THREAD

**Issue by l0rd** [eclipse/che#20622]
### Is your enhancement related to a problem? Please describe

Che-Theia preferences are not persisted anymore. When I refresh the browser, when I restart a workapce ore when I create a new workspace, the preferences are reset.

### Describe the solution you'd like

Persist preferences after refresh/restart and across all user workspaces.

### Release Notes Text

When Che-Theia user preferences (including plugins preferences) are changed, they are persisted in a Kubernetes ConfigMap. When a workspace is restarted or when a new workspace is created the existing user preferences will be loaded. There is still a [bug](https://github.com/eclipse/che/issues/21065) that prevents color theme preferences to be persisted. This is something that already worked whith DevWorkspace disabled.

**Comment by l0rd** [comment]
Related to https://github.com/eclipse-che/che-server/pull/117

**Comment by vinokurig** [comment]
blocked by https://github.com/eclipse/che/issues/20749

**Comment by max-cx** [comment]
Hi, a question to the assignee of this issue:

Will the outcome require any changes to the relevant content of the _Installation Guide_ or _Administration Guide_ or _End-user Guide_?

Yes/No?

**Comment by vinokurig** [comment]
@max-cx 
> Will the outcome require any changes to the relevant content of the Installation Guide or Administration Guide or End-user Guide?

I don't think so, this is just a technical debt.

**Comment by benoitf** [comment]
moving issue to 7.42 milestone as this issue is part of the endgame

**Comment by svor** [comment]
fixed by https://github.com/eclipse-che/che-theia/pull/1260 and https://github.com/devfile/devworkspace-operator/pull/675

**Comment by l0rd** [comment]
This doesn't work for me. I am trying on next channel but preferences are not persisted. I have started a workspace, I set git user name and email, I have changed the theme but when the workspace is restarted these preferences are lost. Reopening.

**Comment by svor** [comment]
for some reason `colorTheme` preferences were overridden after restarting , other preferences where restored without problems:   

https://user-images.githubusercontent.com/1271546/150776894-c3f83b62-a0ec-4cc6-98ff-fb231f80217d.mp4

**Comment by l0rd** [comment]
@svor I forgot to mention that I had setup gitcredentials secret with my github token in my namespace. That may have an effect on git user name and email.

**Comment by svor** [comment]
some investigation around `clolorTheme`: I've tried to set `colorTheme` on old version of Che 7.40.2 and see the same bug that preference is overridden: 

https://user-images.githubusercontent.com/1271546/150781996-d4918ddb-96b5-4f07-a708-c52b8c462874.mp4

**Comment by svor** [comment]
@l0rd @vinokurig I've created another issue about restoring Theme Color preferences #21065  
It's reproducible with older versions of Che and I don't think it's related to the current work.

**Comment by vinokurig** [comment]
Yes, the problem is that `gitconfig` is not stored when workspace stops, but I am not sure if this is related to this issue.

**Comment by l0rd** [comment]
I create a secret:

```
USERNAME=<my-gh-username>
PERSONAL_ACCESS_TOKEN=<my-access-token>
GIT_WEBSITE="github.com"
CREDENTIALS=$(echo -n "https://${USERNAME}:${PERSONAL_ACCESS_TOKEN}@${GIT_WEBSITE}" | base64)  #<== this command works on macOS only

kubectl apply -f - <<EOF
kind: Secret
apiVersion: v1
metadata:
  name: git-credentials-secret
  annotations:
    controller.devfile.io/mount-path: /etc/git-credentials/
  labels:
    controller.devfile.io/watch-secret: 'true'
    controller.devfile.io/git-credential: 'true'
type: Opaque
data:
  credentials: ${CREDENTIALS}
EOF
```

**Comment by vinokurig** [comment]
I tried to create the secret by oAuth flow but this is brocken: https://github.com/eclipse/che/issues/21077
When I tried to manually create the secret and restart the workspace I had such errors:
![screenshot-192 168 64 132 nip io-2022 01 26-18_51_31](https://user-images.githubusercontent.com/7668752/151209074-dc60fba8-c223-43f9-9cb6-2d042209ec2f.png)

**Comment by vinokurig** [comment]
So I managed to add a github token secret to the user namespace, and the `settings.xml` with the git settings was restored after workspace restart.

**Comment by vinokurig** [comment]
The `settings.json` file with git settings was successfully restored on openshift che either.

**Comment by svor** [comment]
@l0rd do you still have this problem? 
we can't reproduce the problem when git preferences are not restored

**Comment by l0rd** [comment]
Let me start a cluster to check that

**Comment by l0rd** [comment]
I was not able to start workspace (https://github.com/eclipse/che/issues/21089). I will try again on Monday.

**Comment by l0rd** [comment]
> @l0rd do you still have this problem?

@svor @vinokurig unfortunately I still have problems although one thing is working:

1. When I restart the workspace after I set the git configuration, the settings.json is persisted 👍 
<img width="389" alt="image" src="https://user-images.githubusercontent.com/606959/152058693-298afec7-d60a-4780-bf11-29422766718f.png">

2. But even if `git.user.name` and `git.user.email` are set in settings.json I still get this message in Theia status bar:
<img width="229" alt="image" src="https://user-images.githubusercontent.com/606959/152058842-62dde420-95b8-436a-a9dc-65796b334fef.png">

3. And the git config is not set in the command line
<img width="242" alt="image" src="https://user-images.githubusercontent.com/606959/152059162-67975a66-92dc-4149-8db0-dc0ee03fb0ba.png">

**Comment by svor** [comment]
@l0rd So, it seems that the problem is not related to the Preferences persisting. The information about username/email in the status bar is taken form the .gitconfig which doesn't exist in the container. The problem with .gitconfig is related to https://github.com/devfile/devworkspace-operator/issues/694, I'm going to close current issue, please reopen it if you don't think so

**Comment by l0rd** [comment]
@svor @vinokurig I have created this separate issue: https://github.com/eclipse/che/issues/21115
